### ISSUE TITLE
Cover: Move color controls to the toolbar

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#20837]
Color controls, like background media, are primary features of this block and should be moved to the toolbar:

![image](https://user-images.githubusercontent.com/548849/76533155-ea1ee300-6477-11ea-8e5b-3a64f373a301.png)

**Comment by ZebulanStanphill** [comment]
Related issue for Group and Columns blocks: #20070.

**Comment by ItsJonQ** [comment]
👋 Halloo! I started looking into this today.

Ma goodness! What a journey. It's definitely a challenge, especially if we want to do it cleanly.

I'm up for taking this one! Just want to leave a note for my future self (and to others)

Don't under estimate this one 🙃

**Comment by ZebulanStanphill** [comment]
@ItsJonQ Yeah, as I discovered in #20246, the way toolbars and popovers relate to each other and how you're supposed to use either is rather complicated. Good luck!

**Comment by ItsJonQ** [comment]
<img width="697" alt="Screen Shot 2020-05-19 at 4 26 58 PM" src="https://user-images.githubusercontent.com/2322354/82374812-95569600-99ed-11ea-96cf-3170fdb91e8f.png">

Got a quick + dirty version working. (Ignore the `W` placeholder Icon for now)

One thing I noticed is that there may be cases where the dropdown menu will cover the content (like above). It's a little unsettling (at least for me). I prefer the controls being on the sidebar (as disconnected as they are). At least it allows me to see the Cover in full view as I adjust it.

If anyone is curious, I pushed up my explorations here:
https://github.com/WordPress/gutenberg/tree/try/cover-color-toolbar-experiment

Branch: `try/cover-color-toolbar-experiment`

**Comment by mtias** [comment]
@ItsJonQ I think this one should be present in both places. There some work left to do to clean the contents but this is a good first step to exposing the most important tools.

**Comment by ItsJonQ** [comment]
@mtias Great! Thanks for your thoughts :). In that case, I'll proceed with this effort!

**Comment by ZebulanStanphill** [comment]
@mtias I thought we wanted to avoid duplicate controls in the toolbar/inspector?

**Comment by apeatling** [comment]
There is an example of doing this in the Navigation block.

**Comment by glendaviesnz** [comment]
Currently the color/gradient controls seem to be very tied to the inspector panels, and they are also in the middle of an update, so it may make sense to defer this until that work is finalised. I have added [a comment over on the color component issue](https://github.com/WordPress/gutenberg/issues/35093#issuecomment-928559912) about this as it would be good to have the use of these components in inspector panels and dropdowns considered as part of that work.

**Comment by glendaviesnz** [comment]
There is a very rudimentary PoC branch that adds the existing color panel to a dropdown just to see how it looks at https://github.com/WordPress/gutenberg/tree/move/cover-color-to-tool-menu

**Comment by mtias** [comment]
Also slightly related: https://github.com/WordPress/gutenberg/pull/34680

**Comment by jasmussen** [comment]
This was a good and valid issue when it was opened, with a design I want to acknowledge. Since then, the Cover block toolbar has grown quite a bit:

<img width="452" height="108" alt="Image" src="https://github.com/user-attachments/assets/5456e5f1-c702-45ff-bfbd-89c989375fd1" />

Other toolbars have grown as well, as have the way colour panels have been organised, splitting into background and typography, and even in the case of Cover, overlay.

Parts of the idea have even been implemented when using the zoomed out view of pattern insertion, where the same icon is used to shuffle colour styles:

<img width="452" height="257" alt="Image" src="https://github.com/user-attachments/assets/99893c1f-3dac-442b-94e3-beda0e78050b" />

Because of this, I'm going to close this issue as wontfix. That doesn't have to be where this ends, but that the implementation has drifted enough that to reopen we'd need a fresh approach and a fresh design. Often such things appear when we close issues, in which case, good! Thank you all for the discussion.
